### PR TITLE
Initial implementation of Fakes framework.

### PR DESCRIPTION
**Pull request by comintern**
ref #2863, #2891 

Consider this WIP.  There are a couple of features that still aren't implemented, and it's only lightly tested at this point. That said, it would be nice to get it merged in that there are quite a few changed files and the merge commit is already promising to be unpleasant.  If push comes to shove, it can easily be "switched off" by making the relevant members `ComVisible(false)`.

### CODE REVIEW

**Comment by retailcoder** [comment]
We need a wiki page to document what the conventions are and how to extend the current functionality.

**Comment by retailcoder** [comment]
@comintern only missing a commit on README.MD and the *about* dialog, to sing the praise of **EasyHook** :-)

**Comment by retailcoder** [comment]
BTW the internal API looks beautiful - feels like extending this is going to be a charm!

File: RetailCoder.VBE/UnitTesting/AssertHandler.cs [hunk 1]
```diff
@@ -10,41 +11,33 @@ public static class AssertHandler
...
-                handler(null, new AssertCompletedEventArgs(TestOutcome.Ignored));
-            }
+            OnAssertCompleted?.Invoke(null, new AssertCompletedEventArgs(TestOutcome.Ignored));
+        }
+
+        [DllImport("vbe7.dll", SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.IUnknown)]
+        private static extern object rtcErrObj();
+
+        public static void RaiseVbaError(int number, string source = "", string description = "", string helpfile = "", int helpcontext = 0)
```
Comment (rubberduck203) [inline RetailCoder.VBE/UnitTesting/AssertHandler.cs]: ++ I don't know why we never thought of importing this before. I just want to point out that, more than ever, we need to stick to our guns about pre-2007 being unsupported.

**Pull request merged**